openstack/imageservice/v2/images/urls.go

```diff
@@ -39,6 +39,14 @@ func deleteURL(c *gophercloud.ServiceClient, imageID string) string {
 
 // builds next page full url based on current url
 func nextPageURL(currentURL string, next string) string {
+	if next == "" {
+		return next
+	}
 	base := currentURL[:strings.Index(currentURL, "/images")]
+	// The next URL may contain the version (i.e., /v2/images). Exclude it if so.
+	idx := strings.Index(next, "/images")
+	if idx != -1 {
+		next = next[idx:]
```

Comment:
Any need to worry about training slash or no trailing slash, or is this already taken care of?